plugins/Marketplace/Controller.php

```diff
@@ -477,11 +477,19 @@ private function createUpdateOrInstallView($template, $nonceName)
             try {
                 $this->pluginInstaller->installOrUpdatePluginFromMarketplace($pluginName);
             } catch (\Exception $e) {
-                $notification = new Notification($e->getMessage());
+                $message = $e->getMessage();
+                $isRaw = false;
+                if (stripos($message, 'PCLZIP_ERR_BAD_FORMAT') !== false) {
```

Comment:
We are catching a quite generic error message here and are printing an error message that only troubleshoots for premium plugins.
Isn't there a better way to handle that more specific? If I understood it correctly the problem mainly occurs when a premium plugin doesn't contain a download link. Such the download fails and the PCLZIP error is thrown as it tries to extract an non existing or empty file.

I guess the problem already happens here:
https://github.com/matomo-org/matomo/blob/9a3ef94df61682e7d543f3f8b438c368f6eb71a6/plugins/CorePluginsAdmin/PluginInstaller.php#L64

Following this it triggers a download through the marketplace client:

https://github.com/matomo-org/matomo/blob/9a3ef94df61682e7d543f3f8b438c368f6eb71a6/plugins/Marketplace/Api/Client.php#L139-L163

As you can see, if there is no download url, or it fails to download for any reason, the method simply returns `false`.
But this `false` isn't handled anywhere, so it tries to proceed as if the download would have happened.

I think it would be a smarter solution to adjust the code above, so it throws an exception when it isn't able to find or download something. If that happens you could check if the plugin is a premium plugin, and only in that case print the below error message.

Or did I maybe miss anything? Haven't spent too much time investigating that tbh.

Reply:
@sgiehl We added the code to not show the install button when there is no download link, but we have a report from a user who has a valid license and the download URL and they still see this error, hence we added this extra message so that users can proceed with a manual installation atleast

Reply:
Ah ok. But then we should not show a premium plugin related message if the plugin isn't one

Reply:
@sgiehl Updated the code to show a different message for free plugins, good catch :+1: